### PR TITLE
number, string: add S.min and S.max

### PR DESCRIPTION
**Pull request by davidchambers**
```haskell
min :: Ord a => a -> a -> a
max :: Ord a => a -> a -> a
```

[`R.min`][1] and [`R.max`][2], but with type checking. I want to tick some boxes in #145. :)

[1]: http://ramdajs.com/docs/#min
[2]: http://ramdajs.com/docs/#max

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -169,6 +169,16 @@
...
+  //  Ord :: TypeClass
+  var Ord = $.TypeClass(
+    'sanctuary/Ord',
+    function(x) {
+      var type = _type(x);
+      return type === 'Number' && $.ValidNumber.test(x) ||
+             type === 'String';
+    }
+  );
+
```
Comment (davidchambers) [inline index.js]: We should document all the type classes at some point.

File: test/index.js [hunk 2]
```diff
@@ -3575,6 +3575,80 @@ describe('number', function() {
...
+
+    it('may be applied to (valid) numbers', function() {
+      eq(S.min(42, 99), 42);
+      eq(S.min(0.1, 0.01), 0.01);
+      eq(S.min(Infinity, -Infinity), -Infinity);
+    });
+
+    it('may be applied to strings', function() {
+      eq(S.min('abc', 'xyz'), 'abc');
+      eq(S.min('xyz', 'abc'), 'abc');
```
Comment (svozza) [inline test/index.js]: Maybe add a case sensitive test?

File: index.js (hunk 1 above)
Comment (svozza) [inline index.js]: Agreed. This will also be a very useful type.

File: test/index.js (hunk 2 above)
Comment (davidchambers) [inline test/index.js]: :zap:

**Comment by davidchambers** [comment]
Prompted by your comment, @svozza, I added several more assertions. I also updated the examples to highlight the difference between number comparison and string comparison, and added a paragraph to each function's description to explain the distinction.

**Comment by svozza** [comment]
LGTM.

**Pull request merged**